network/src/commonMain/kotlin/com/wire/kalium/network/api/v3/authenticated/MessageApiV3.kt

```diff
@@ -20,9 +20,39 @@ package com.wire.kalium.network.api.v3.authenticated
 
 import com.wire.kalium.network.AuthenticatedNetworkClient
 import com.wire.kalium.network.api.base.authenticated.message.EnvelopeProtoMapper
+import com.wire.kalium.network.api.base.authenticated.message.MessageApi
+import com.wire.kalium.network.api.base.authenticated.message.QualifiedSendMessageResponse
+import com.wire.kalium.network.api.base.model.ConversationId
 import com.wire.kalium.network.api.v2.authenticated.MessageApiV2
+import com.wire.kalium.network.exceptions.ProteusClientsChangedError
+import com.wire.kalium.network.serialization.XProtoBuf
+import com.wire.kalium.network.utils.NetworkResponse
+import com.wire.kalium.network.utils.wrapKaliumResponse
+import io.ktor.client.call.body
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
 
 internal open class MessageApiV3 internal constructor(
-    authenticatedNetworkClient: AuthenticatedNetworkClient,
-    envelopeProtoMapper: EnvelopeProtoMapper
-) : MessageApiV2(authenticatedNetworkClient, envelopeProtoMapper)
+    private val authenticatedNetworkClient: AuthenticatedNetworkClient,
+    private val envelopeProtoMapper: EnvelopeProtoMapper
+) : MessageApiV2(authenticatedNetworkClient, envelopeProtoMapper) {
+
+    override suspend fun qualifiedSendMessage(
```

Comment:
Why do you need to override this implementation? Isn't it the same as V0?

Reply:
You are right, I was going to implement some changes related to handling this new field, but I'll do it in another PR/ticket 👀